doc/api/embedding.md

```diff
@@ -167,8 +167,1327 @@ int RunNodeInstance(MultiIsolatePlatform* platform,
+
+This is an opaque pointer for the Node-API scope where the provided `napi_env`
+is valid.
+
+#### Callback types
+
+##### `node_embedding_run_node_api_callback`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+```c
+typedef void(NAPI_CDECL* node_embedding_run_node_api_callback)(
+    void* cb_data,
+    napi_env env);
+```
+
+Function pointer type for callback that invokes Node-API code.
+
+The callback parameters:
+
+- `[in] cb_data`: The user data associated with this callback.
+- `[in] env`: Node-API environment.
+
+#### Functions
+
+##### `node_embedding_runtime_node_api_run`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+Runs Node-API code.
+
+```c
+node_embedding_status NAPI_CDECL
+node_embedding_runtime_node_api_run(
+    node_embedding_runtime runtime,
+    node_embedding_node_api_run_callback run_node_api,
+    void* run_node_api_data);
+```
+
+- `[in] runtime`: The Node.js embedding_runtime instance.
+- `[in] run_node_api`: The callback to invoke Node-API code.
+- `[in] run_node_api_data`: Optional. The data associated with the
+  `run_node_api` callback.
+
+Returns `node_embedding_status_ok` if there were no issues.
+
+The function invokes the callback that runs Node-API code in the Node-API scope.
+It triggers the uncaught exception handler if there were any unhandled
+JavaScript errors.
+
+##### `node_embedding_runtime_node_api_scope_open`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+Opens scope to run Node-API code.
+
+```c
+node_embedding_status NAPI_CDECL
+node_embedding_runtime_node_api_scope_open(
+    node_embedding_runtime runtime,
+    node_embedding_node_api_scope* node_api_scope,
+    napi_env* env);
+```
+
+- `[in] runtime`: The Node.js embedding_runtime instance.
+- `[out] node_api_scope`: The opened Node-API scope.
+- `[out] env`: The Node-API environment that can be used in the scope.
+
+Returns `node_embedding_status_ok` if there were no issues.
+
+The function opens up V8 Isolate, V8 handle, and V8 context scopes where it is
+safe to run the Node-API environment.
+
+Using the `node_embedding_runtime_node_api_scope_open` and `node_embedding_runtime_node_api_scope_close` is an alternative to the
+`node_embedding_runtime_node_api_run` call.
+
+##### `node_embedding_runtime_node_api_scope_close`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+Closes the Node-API invocation scope.
+
+```c
+node_embedding_status NAPI_CDECL
+node_embedding_runtime_node_api_scope_close(
+    node_embedding_runtime runtime,
+    node_embedding_node_api_scope node_api_scope);
+```
+
+- `[in] runtime`: The Node.js embedding_runtime instance.
+- `[in] node_api_scope`: The Node-API scope to close.
+
+Returns `node_embedding_status_ok` if there were no issues.
+
+The function closes the V8 Isolate, V8 handle, and V8 context scopes.
+Then, it triggers the uncaught exception handler if there were any
+unhandled errors.
+
```

Comment:
Just be before the examples it might be useful to show the steps to get javascript run
I assume something like
* create platform
* create runtime
..etc

Reply:
This a great suggestion! As soon as I started to work on the description I see that I must new APIs for consistency.